cli/internal/run/dry_run.go

```diff
@@ -173,6 +175,43 @@ func executeDryRun(ctx gocontext.Context, engine *core.Engine, g *graph.Complete
 	return taskIDs, nil
 }
 
+func populateCacheState(turboCache cache.Cache, taskSummaries []*taskSummary) error {
+	// We make at most 8 requests at a time for cache state.
+	maxParallelRequests := 8
+	taskCount := len(taskSummaries)
+
+	parallelRequestCount := maxParallelRequests
+	if taskCount < maxParallelRequests {
+		parallelRequestCount = taskCount
+	}
+
+	queue := make(chan int, taskCount)
+
+	var returnErr error
+
+	wg := &sync.WaitGroup{}
+	for i := 0; i < parallelRequestCount; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for index := range queue {
+				task := taskSummaries[index]
+				itemStatus, err := turboCache.Exists(task.Hash)
+				task.CacheState = itemStatus
```

Comment:
Should we only be altering the cache state if there's an error checking the cache?

Reply:
The codepath underneath the hood guarantees `ItemStatus{}, error`, so it's safe to set... but also we blow up the entire dry run if there _is_ an error, so it's fine.

Reply:
Arguably, `Exists` throwing an error should just mean `false`; always assuming that the error is repeatable with `false` being a valid outcome.

Reply:
We only call `cache.Exists` in `dry_run.go`. It could map an error to `false`, which is conservative, doesn't unnecessarily blow up a dry run call, and would provide the same expected outcome:

Anyway, here is that alternative:
https://github.com/nathanhammond/turbo/pull/7/files